pallets/liquidity-pools/src/lib.rs

```diff
@@ -689,6 +685,51 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Collect a user's foreign investment as if we had received a
+		/// `CollectInvest` message from another domain.
+		#[pallet::call_index(10)]
+		#[pallet::weight(10_000 + T::DbWeight::get().writes(1).ref_time())]
+		pub fn collect_foreign_investment_for(
```

Comment:
NOTE: Required for collector bot. In the future, this should be handled inherently, i.e. in `on_initialize` or `on_idle`.